test/iobuffer.jl

```diff
@@ -497,6 +497,13 @@ end
     end
 end
 
+@testset "issue #57962" begin
+    io = IOBuffer(repeat("x", 400))
+    skip(io, 10)
+    skip(io, 400)
+    @test position(io) == 400
```

Comment:
Note that this result is a bug also (correct answer is 410), and any package relying on this is going to have a bad time

Reply:
TranscodingStreams.jl has it's own more specific definition of `skip` so doesn't rely on this.
https://github.com/JuliaIO/TranscodingStreams.jl/blob/ea4533afaf175443198f5d98009266b6b5f2d882/src/stream.jl#L355-L362

Reply:
I changed this test to instead check `isempty(read(io))` after skipping past the end. This seems consistent with how `IOStream` currently works as well. Though currently https://github.com/JuliaLang/julia/pull/58024 says this should throw some kind of error.